src/validations.ts

```diff
@@ -26,25 +26,39 @@ export function validateRequiredField(
 
 export function validatePassword(
   value: string,
-): { empty: true } | { charLength?: boolean; digit?: boolean } | null {
+): { empty: true } | { charLength?: boolean; digit?: boolean; letter?: boolean } | null {
   const digitRegex = /[0-9]/;
+  const letterRegex = /[a-zA-Z]/;
 
   if (!value) {
     return { empty: true };
   }
 
   const charError = value.length < 8;
   const digitError = !digitRegex.test(value);
+  const letterError = !letterRegex.test(value);
 
+  if (charError && digitError && letterError) {
+    return { charLength: true, digit: true, letter: true };
+  }
   if (charError && digitError) {
     return { charLength: true, digit: true };
   }
+  if (charError && letterError) {
+    return { charLength: true, letter: true };
+  }
+  if (digitError && letterError) {
+    return { digit: true, letter: true };
+  }
```

Comment:
Is it not possible to simplify this code in the following way:

```js
if (charError || digitError || letterError) {
    return { charLength: charError, digit: digitError, letter: letterError };
}
```

Reply:
In place of: 

```js
  if (charError && digitError && letterError) {
    return { charLength: true, digit: true, letter: true };
  }
  if (charError && digitError) {
    return { charLength: true, digit: true };
  }
  if (charError && letterError) {
    return { charLength: true, letter: true };
  }
  if (digitError && letterError) {
    return { digit: true, letter: true };
  }
  if (charError) {
    return { charLength: true };
  }
  if (digitError) {
    return { digit: true };
  }
  if (letterError) {
    return { letter: true };
  }
```

Reply:
Changed.